python/lsst/meas/deblender/baseline.py

```diff
@@ -33,50 +34,232 @@
+        self.debResult = debResult
+        self.peakCount = debResult.peakCount
         self.templateSum = None
+        
+        # avgNoise is an estiamte of the average noise level for the image in this filter
+        if avgNoise is None:
+            stats = afwMath.makeStatistics(self.varimg, self.mask, afwMath.MEDIAN)
+            avgNoise = math.sqrt(stats.getValue(afwMath.MEDIAN))
+            debResult.log.trace('Estimated avgNoise for filter %s = %f', self.filter, avgNoise)
+        self.avgNoise = avgNoise
+        
+        # Store all of the peak information in a single object
+        self.peaks = []
+        peaks = self.fp.getPeaks()
+        for idx in range(self.peakCount):
+            deblendedPeak = DeblendedPeak(peaks[idx], idx, self)
+            self.peaks.append(deblendedPeak)
+
+    def updateFootprintBbox(self):
+        """Update the bounding box of the parent footprint
+        
+        If the parent Footprint is resized, for example when `meas.deblender.baseline.deblend`
+        has `patchEdges=True`, it will be necessary to update the bounding box of the footprint.
+        """
+        # Pull out the image bounds of the parent Footprint
+        self.imbb = self.img.getBBox()
+        self.bb = self.fp.getBBox()
+        if not self.imbb.contains(self.bb):
+            raise ValueError(('Footprint bounding-box %s extends outside image bounding-box %s') %
+                             (str(self.bb), str(self.imbb)))
+        self.W, self.H = self.bb.getWidth(), self.bb.getHeight()
+        self.x0, self.y0 = self.bb.getMinX(), self.bb.getMinY()
+        self.x1, self.y1 = self.bb.getMaxX(), self.bb.getMaxY()
+
+class MultiColorPeak(object):
+    """Collection of single peak deblender results in multiple bands.
+    
+    There is one of these objects for each Peak in the footprint.
+    """
+    
+    def __init__(self, peaks, pki, parent):
+        """Create a collection for deblender results in each band.
+        
+        Parameters
+        ----------
+        peaks: `dict` of `afw.detection.PeakRecord`
+            Each entry in ``peaks`` is a peak record for the same peak in a different band
+        pki: int
+            Index of the peak in `parent.peaks`
+        parent: `DeblendedParent`
+            DeblendedParent object that contains the peak as a child.
+        
+        Returns
+        -------
+        None
+        
+        """
+        self.filters = list(peaks.keys())
+        self.deblendedPeaks = peaks
+        self.parent = parent
+        for pki, peak in self.deblendedPeaks.items():
+            peak.multiColorPeak = self
+
+        # Fields common to the peak in all bands that will be set by the deblender
+        # In the future this is likely to contain information about the probability of the peak
+        # being a point source, color-information about templateFootprints, etc.
+        self.pki = pki
+        self.skip = False
+        self.deblendedAsPsf = False
```

Comment:
Maybe. I think it would be best to add extra members as needed when we add features as opposed to guessing what we might need and end up with a lot of unused parameters.